### ISSUE TITLE
Support multi-dim reductions for torch.prod, torch.all, torch.any

### ISSUE THREAD

**Issue by vishwakftw** [pytorch/pytorch#56586]
## 🚀 Feature

Currently, only `torch.sum` and `torch.mean` have multi-dimension reductions. It would be nice to extend this property to reduction operations such as `torch.prod`, `torch.all` and `torch.any`. This will also ensure that the interface is similar to NumPy, which supports this.

The current alternative to doing multi-dim reductions is to chain the operations like so:
```python
import torch

x = torch.randn(2, 3, 5)
x.prod(-1).prod(-1)  # can be replaced by x.prod(dim=(-1, -2))

y = torch.randint(2, (2, 3, 5), dtype=bool)
y.all(-1).all(-1)  # can be replaced by y.all(dim=(-1, -2))
# analogously for torch.any
```

**Comment by harishsdev** [comment]
As per documentation we have below functionality

torch.mean(input, dim, keepdim=False, *, out=None) → Tensor
torch.prod(input, dim, keepdim=False, *, dtype=None) → Tensor
torch.sum(input, dim, keepdim=False, *, dtype=None) → Tensor

Can you please elaborate what is the requirement

**Comment by vishwakftw** [comment]
The `dim` argument in `prod` only accepts an integer, whereas in `mean` and `sum` it accepts an integer or a tuple of integers. Refer the Python comments in the chaining example above.
